packages/tablev2/Table.tsx

```diff
@@ -0,0 +1,265 @@
+  const onClick =
+    config.sorter && (() => update({ sortBy: id, order: nextDir }));
+  const ariaSortStringMap: { [key: string]: "ascending" | "descending" } = {
+    asc: "ascending",
+    desc: "descending"
+  };
+  const HeaderCellContent = () => (
+    <>
+      <div className={cx(sortClass, textTruncate)}>{config.header}</div>
+      <Draggable onRelativeXChange={onRelativeXChange} />
+    </>
+  );
+
+  return (
+    <div
+      className={style.headerCell}
+      id={id}
+      ref={cell}
+      role="columnheader"
+      aria-sort={order ? ariaSortStringMap[order] : "none"}
+    >
+      {config.sorter ? (
+        <ResetButton onClick={onClick}>
+          <HeaderCellContent />
+        </ResetButton>
+      ) : (
+        <HeaderCellContent />
+      )}
+    </div>
+  );
+}
+
+type HeaderRowProps<Entry> = {
+  columns: TableProps<Entry>["columns"];
+  stickyFirstCol: TableProps<Entry>["stickyFirstCol"];
+  state: State;
+  update: (a: Partial<State>) => void;
+};
+function HeaderRow<Entry>({
+  columns,
+  state,
+  stickyFirstCol,
+  update
+}: HeaderRowProps<Entry>) {
+  const toHeaderCell = ({ id }) => {
+    return (
+      <HeaderCell
+        key={id}
+        id={id}
+        config={columns.find(c => c.id === id)!}
+        update={update}
+        state={state}
+      />
+    );
+  };
+
+  return (
+    <div
+      role="row"
+      className={cx(
+        getCellClassName(
+          state.columns.map(c => c.width),
+          stickyFirstCol ?? true
+        ),
+        style.headerRow
+      )}
+      key="headerRow"
+    >
+      {state.columns.map(toHeaderCell)}
+    </div>
+  );
+}
+
+type DivProps = React.HTMLAttributes<HTMLDivElement>;
+export function Table<Entry, ColIDs extends string, Sort extends ColIDs>({
+  data,
+  columns,
+  initialSorter,
+  onStateChange,
+  stickyFirstCol,
+  toId,
+  ...divProps
+}: { data: Entry[] } & DivProps & TableProps<Entry, ColIDs, Sort>) {
+  const [state, setState] = React.useState<State>({
+    columns: columns.map(c => ({
+      id: c.id,
+      width: c.initialWidth || "1fr"
+    })),
+    order: initialSorter?.order || "asc",
+    sortBy: initialSorter?.by || null
+  });
+
+  const update = (x: Partial<State>) => {
+    setState({ ...state, ...x });
+    onStateChange?.({ ...state, ...x });
+  };
+  const col = columns.find(c => c.id === state.sortBy);
+  const sort = col?.sorter?.(state.order) || (() => 0);
+
+  // We currently need to cast Config to Config<unknown> as to a bug in typescript's handling of unknowns. We actually NARROW the interface, which is good.
+  return (
+    <div {...divProps} className={style.table} role="grid">
+      <HeaderRow
+        columns={columns}
+        stickyFirstCol={stickyFirstCol}
+        state={state}
+        update={update}
+      />
+
+      {data.sort(sort).map(el => (
+        <Row
+          key={"row-" + toId(el)}
+          // TODO: figure out how to pass a generic while using React.memo
+          // @ts-ignore
```

Comment:
Mighty Typescript wizard @pierrebeitz - any ideas? :)

Reply:
The issue is no more! types should look much more documenting now.